topi/python/topi/nn/conv2d.py

```diff
@@ -265,6 +267,22 @@ def conv2d_nchw(Input, Filter, stride, padding, out_dtype='float32'):
         stride_h = stride_w = stride
     else:
         stride_h, stride_w = stride
+    if isinstance(padding, int):
+        pad_h = pad_w = padding
+    else:
+        pad_h, pad_w = padding
+
+    target = tvm.target.current_target()
+    if target and target.target_name == "rocm" and "miopen" in target.libs:
```

Comment:
since the logic is rocm specific, instead of putting it here, move it to topi/rocm and use generic dispatch mechanism to trigger this as in schedule

Reply:
done